File: app/src/main/java/com/example/roomres/MODELS/Reservation.java
package com.example.roomres.MODELS;

import androidx.annotation.NonNull;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;

public class Reservation implements Serializable, Comparable<Reservation> {

    @SerializedName("id")
    private int id;

    @SerializedName("fromTime")
    private int fromTime;

    @SerializedName("toTime")
    private int toTime;

    @SerializedName("userId")
    private String userId;

    @SerializedName("purpose")
    private String purpose;

    @SerializedName("roomId")
    private int roomId;

    private DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd hh:mm");

    public Reservation(){}

    public Reservation(int id, int fromTime, int toTime, String userId, String purpose, int roomId) {
        this.id = id;
        this.fromTime = fromTime;
        this.toTime = toTime;
        this.userId = userId;
        this.purpose = purpose;
        this.roomId = roomId;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getFromTimeOld() {
        return fromTime;
    }

    public void setFromTime(int fromTime) {
        this.fromTime = fromTime;
    }

    public int getToTimeOld() {
        return toTime;
    }

    public void setToTime(int toTime) {
        this.toTime = toTime;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getPurpose() {
        return purpose;
    }

    public void setPurpose(String purpose) {
        this.purpose = purpose;
    }

    public int getRoomId() {
        return roomId;
    }

    public void setRoomId(int roomId) {
        this.roomId = roomId;
    }

    public Date getFromTime() {
        return new Date(getFromTimeOld()*1000);
    }

    public Date getToTime() {
        return new Date(getToTimeOld()*1000);
    }

    @Override
    public int compareTo(Reservation reservation) {
        return (this.getToTimeOld() < reservation.getToTimeOld() ? -1 :
                (this.getToTimeOld() == reservation.getToTimeOld() ? 0 : 1));
    }

    @NonNull
    @Override
    public String toString() {
        return "Purpose: "+ purpose + "\nFrom: " + dateFormat.format(getFromTime()) + "\nTo: " + dateFormat.format(getToTime());
    }



}

File: app/src/main/java/com/example/roomres/MODELS/Reservation.kt
package com.example.roomres.MODELS

import com.google.gson.annotations.SerializedName

import java.io.Serializable
import java.text.DateFormat
import java.text.SimpleDateFormat
import java.util.Date

class Reservation : Serializable, Comparable<Reservation> {

    @SerializedName("id")
    var id: Int = 0

    @SerializedName("fromTime")
    var fromTimeOld: Int = 0
        private set

    @SerializedName("toTime")
    var toTimeOld: Int = 0
        private set

    @SerializedName("userId")
    var userId: String? = null

    @SerializedName("purpose")
    var purpose: String? = null

    @SerializedName("roomId")
    var roomId: Int = 0

    private val dateFormat = SimpleDateFormat("yyyy-MM-dd hh:mm")

    val fromTime: Date
        get() = Date((fromTimeOld * 1000).toLong())

    val toTime: Date
        get() = Date((toTimeOld * 1000).toLong())

    constructor() {}

    constructor(id: Int, fromTime: Int, toTime: Int, userId: String, purpose: String, roomId: Int) {
        this.id = id
        this.fromTimeOld = fromTime
        this.toTimeOld = toTime
        this.userId = userId
        this.purpose = purpose
        this.roomId = roomId
    }

    fun setFromTime(fromTime: Int) {
        this.fromTimeOld = fromTime
    }

    fun setToTime(toTime: Int) {
        this.toTimeOld = toTime
    }

    override fun compareTo(reservation: Reservation): Int {
        return if (this.toTimeOld < reservation.toTimeOld)
            -1
        else
            if (this.toTimeOld == reservation.toTimeOld) 0 else 1
    }

    override fun toString(): String {
        return "Purpose: " + purpose + "\nFrom: " + dateFormat.format(fromTime) + "\nTo: " + dateFormat.format(toTime)
    }


}
